Jenkinsfile

```diff
@@ -121,11 +126,14 @@ node("headless-macos") {
 
     processes["headless-and-archive"] = {
       stage("headless tests") {
-        // Prevent multiple headless tests running at once
-        lock("headless tests") {
-          // chrome-headless tests run on the same node as the pipeline
-          // we are re-using the state prepared by `ci-all` below
-          // if we ever change these tests to run on a different node, rollup is required in addition to the normal CI command
+        // TODO: Determine if this should be run on a container instead
```

Comment:
Where we do any of this will depend on the performance. We started using headless macos because it was way, way faster.

You'll still need the lock, inside the `headless-macos` node, because it has multiple executors. Or change it to not use general macos, not headless macos, for these tests.

Reply:
Used lock for `headless-macos`